model/categories.model.js

```diff
@@ -70,22 +59,59 @@ exports.updateReviewById = async (review_id, inc_votes) => {
 
     const [review] = rows;
     if (review) return { review };
-    return Promise.reject({
-      status: 404,
-      msg: `There is no review with the id ${review_id} to update`,
-    });
+    return resourceError(review_id, "review");
+  } catch (err) {
+    return dbError(err, `${inc_votes} is not a valid number of votes`);
+  }
+};
+
+exports.selectCommentsByReviewId = async (review_id) => {
+  try {
+    const {
+      rows: [review],
+    } = await pool.query(
+      `
+      SELECT (
+        CASE WHEN EXISTS 
+        (SELECT 1 FROM reviews WHERE review_id = $1)
+        THEN true ELSE false END
+      ) as exists
+      `,
+      [review_id]
+    );
+    if (!review.exists) return resourceError(review_id, "review");
+
+    const { rows } = await pool.query(
+      `
+      SELECT * FROM comments
+      WHERE review_id = $1
+      `,
+      [review_id]
+    );
+    return rows;
   } catch (err) {
-    if (err.code === "22P02") {
-      return Promise.reject({
-        status: 400,
-        msg: `${inc_votes} is not a valid number of votes`,
-      });
-    }
-    return Promise.reject();
+    return dbError(err, `${review_id} is not a valid review id`);
   }
 };
 
 exports.selectUsers = async () => {
   const { rows } = await pool.query("SELECT * FROM users");
   return rows;
 };
+
+const resourceError = (id, type) => {
+  return Promise.reject({
+    status: 404,
+    msg: `There is no ${type} with the id ${id}`,
+  });
+};
+
+const dbError = (err, msg) => {
+  if (err.code === "22P02") {
+    return Promise.reject({
+      status: 400,
+      msg,
+    });
+  }
+  return Promise.reject();
+};
```

Comment:
these 2 error handling functions should be in a separate error handling file/folder please